### PR TITLE
[SeaTunnel #727][Core] Split seatunnel-core into spark and flink child module

### LINKED ISSUE #727: [Feature][Core] Split seatunnel-core into spark and flink child module
### Search before asking

- [X] I had searched in the [feature](https://github.com/InterestingLab/seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

In seatunnel project, the `seatunnel-core` module includes `spark`  and `flink`, it will cause the `pom.xml` contains many dependencies, they may affect each other. So we can split `seatunnel-core` into two more independent sub module, like
```
seatunnel-core
- seatunnel-flink-core
- seatunnel-spark-core
```

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by leo65535**
Fixed #727 

## Purpose of this pull request

In seatunnel project, the `seatunnel-core` module includes `spark`  and `flink`, it will cause the `pom.xml` contains many dependencies, they may affect each other. So we can split `seatunnel-core` into two more independent sub module, like
```
seatunnel-core
├── pom.xml
├── seatunnel-core-base
├── seatunnel-core-flink
└── seatunnel-core-spark

[seatunnel-dist]$ tree seatunnel-dist-2.0.4-2.11.8/lib/ -L 1
seatunnel-dist-2.0.4-2.11.8/lib/
├── seatunnel-core-flink-2.0.4-2.11.8.jar
└── seatunnel-core-spark-2.0.4-2.11.8.jar
```

## Check

Test the flink task manually.

image

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by CalvinKirs** [comment]
@RickyHuo @garyelephant @kid-xiong WDYT?

File: seatunnel-core/seatunnel-core-spark/pom.xml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        <dependency>
+            <groupId>com.github.scopt</groupId>
+            <artifactId>scopt_2.11</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>io.github.interestinglab.seatunnel</groupId>
+            <artifactId>seatunnel-spark-api</artifactId>
+            <version>${project.version}</version>
+            <!--<scope>provided</scope>-->
```
Comment (kezhenxu94) [inline seatunnel-core/seatunnel-core-spark/pom.xml:48]: nit: let's remove these commented lines

File: seatunnel-core/seatunnel-core-spark/pom.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            <groupId>io.github.interestinglab.seatunnel</groupId>
+            <artifactId>plugin-spark-sink-console</artifactId>
+            <version>${project.version}</version>
+            <!--<scope>provided</scope>-->
+        </dependency>
+        <dependency>
+            <groupId>io.github.interestinglab.seatunnel</groupId>
+            <artifactId>plugin-spark-sink-elasticsearch</artifactId>
+            <version>${project.version}</version>
+            <scope>compile</scope>
```
Comment (kezhenxu94) [inline seatunnel-core/seatunnel-core-spark/pom.xml:126]: Let's remove these `<scope>compile</scope>` lines, as the default scope is already `compile`

**Review by kezhenxu94 — COMMENTED** [review]
Just some nits, others look good to me 💯

**Comment by yx91490** [comment]
I think there is no need to split the core, it'll make the project more complex. otherwise we should remove plugin dependencies from the core.

File: seatunnel-core/seatunnel-core-spark/pom.xml (hunk 2 above)
Comment (leo65535) [reply seatunnel-core/seatunnel-core-spark/pom.xml]: make sense.

**Comment by timothy65535** [comment]
+1, I think these change is very useful. LGTM

**Comment by leo65535** [comment]
> I think there is no need to split the core

hi, it's necessary to let `spark` and `flink` be independent of each other from a long-term perspective. 

In fact, we already met the bug which caused by dependency, please go to

- https://github.com/InterestingLab/seatunnel/pull/723

**Comment by leo65535** [comment]
cc @CalvinKirs @kezhenxu94, thanks for the review, patch is updated accordingly, can you please take another look?

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Pull request merged**